rasa/core/policies/ted_policy.py

```diff
@@ -755,39 +812,110 @@ def _emebed_dialogue(
     def _encode_features_per_attribute(
         self, tf_batch_data: Dict[Text, Dict[Text, List[tf.Tensor]]], attribute: Text
     ) -> Optional[tf.Tensor]:
-        """
-        Encodes features for a given attribute
+        """Encodes features for a given attribute
+
         Args:
             tf_batch_data: dictionary mapping every attribute to its features and masks
-            attribute: the attribute we will encode features for (e.g., ACTION_NAME, INTENT)
+            attribute: the attribute we will encode features for
+            (e.g., ACTION_NAME, INTENT)
+
         Returns:
             A tensor combining  all features for `attribute`
         """
 
-        if not tf_batch_data[attribute]:
-            return None
-
         attribute_mask = tf_batch_data[attribute][MASK][0]
-        # TODO transformer has to be used to process sequence features
-        attribute_features = self._combine_sparse_dense_features(
-            tf_batch_data[attribute][SENTENCE],
-            f"{attribute}_{SENTENCE}",
-            mask=attribute_mask,
-        )
 
-        if attribute in FEATURES_TO_ENCODE + LABEL_FEATURES_TO_ENCODE:
-            attribute_features = self._tf_layers[f"ffnn.{attribute}_{SENTENCE}"](
+        if attribute in SEQUENCE_FEATURES_TO_ENCODE:
+            _sequence_lengths = tf.cast(
+                tf_batch_data[attribute][SEQUENCE_LENGTH][0], dtype=tf.int32
+            )
+            mask_sequence_text = self._compute_mask(tf.squeeze(_sequence_lengths))
+            sequence_lengths = _sequence_lengths + 1
+            mask_text = self._compute_mask(tf.squeeze(sequence_lengths))
+
+            attribute_features, _, _, _ = self._create_sequence(
+                tf_batch_data[attribute][SEQUENCE],
+                tf_batch_data[attribute][SENTENCE],
+                mask_sequence_text,
+                mask_text,
+                attribute,
+                sparse_dropout=self.config[SPARSE_INPUT_DROPOUT],
+                dense_dropout=self.config[DENSE_INPUT_DROPOUT],
+                masked_lm_loss=self.config[MASKED_LM],
+                sequence_ids=False,
+            )
+
+            # TODO entities
+
+            # resulting attribute features will have shape
+            # combined batch dimension and dialogue length x sequence length x units
+            attribute_features = self._last_token(
+                attribute_features, tf.squeeze(sequence_lengths)
+            )
+
+        else:
+            # resulting attribute features will have shape
+            # combined batch dimension and dialogue length x 1 x units
+            attribute_features = self._combine_sparse_dense_features(
+                tf_batch_data[attribute][SENTENCE],
+                f"{attribute}_{SENTENCE}",
+                mask=attribute_mask,
+            )
+
+        if attribute in SENTENCE_FEATURES_TO_ENCODE + LABEL_FEATURES_TO_ENCODE:
```

Comment:
I think we still need to do that for sequence features because we convert from transformer units (128) to encoding units (50), or remove it all together and transform everything into 128

Reply:
`SENTENCE_FEATURES_TO_ENCODE` includes the `SEQUENCE_FEATURES_TO_ENCODE`, but will update the check to make it more explicit